### PR TITLE
fea/schema v2.0

### PR DESCRIPTION
**Pull request by mimir-d**
- rewrite the whole schema for v2.0
- various fixes for validator; unit tests work

### CODE REVIEW

File: json_spec/output/root.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/opencomputeproject/ocp-diag-core/output",
+    "title": "OutputArtifact",
+    "description": "TODO: link to spec md link",
+    "properties": {
+        "sequenceNumber": {
+            "description": "TODO: link to spec md link",
+            "type": "integer",
+            "minimum": 0,
+            "maximum": 2147483647
```
Comment (dhawkes) [inline json_spec/output/root.json:11]: We didn't bother specifying integer overflow for the other integers defined in this spec. Should be consistent and either remove this or add it for the other indices.

File: json_spec/output/test_step_artifact.json [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                    "$ref": "#/$defs/extension"
+                }
+            },
+            "required": [
+                "extension"
+            ]
+        }
+    ],
+    "$defs": {
+        "file": {
```
Comment (dhawkes) [inline json_spec/output/test_step_artifact.json:138]: Should we pull this out into it's own file (and others like it) to better match the spec doc even if it isn't referenced elsewhere?

**Review by dhawkes — CHANGES_REQUESTED** [review]

File: json_spec/output/dut_info.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/opencomputeproject/ocp-diag-core/dutInfo",
+    "title": "Dut info",
+    "description": "TODO: link to md doc",
+    "properties": {
+        "id": {
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply json_spec/output/dut_info.json:7]: yes, that change happened after i posted the PR; will update

File: json_spec/output/dut_info.json [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            "description": "TODO: link to md doc",
+            "type": "string"
+        },
+        "hostname": {
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply json_spec/output/dut_info.json:11]: same as the prev comment; will update

File: json_spec/output/test_step_artifact.json (hunk 2 above)
Comment (mimir-d) [reply json_spec/output/test_step_artifact.json:138]: ive considered that, but generally stuck to the rule of "referenced in more than 1 place". let me see how it looks as a different file and i'll update.

File: json_spec/output/dut_info.json [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                    "type": "string"
+                }
+            },
+            "required": [
+                "info"
+            ],
+            "additionalProperties": false
+        },
+        "softwareInfo": {
+            "properties": {
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply json_spec/output/dut_info.json:59]: seems i just moved metadata at the end. makes more sense to me as an "extra" thing. i'll change the order in the MD as well

File: json_spec/output/dut_info.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mimir-d) [reply json_spec/output/dut_info.json]: done, please close after check

File: json_spec/output/dut_info.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mimir-d) [reply json_spec/output/dut_info.json]: done, please close after check

File: json_spec/output/root.json (hunk 1 above)
Comment (mimir-d) [reply json_spec/output/root.json]: done, please close after check

File: json_spec/output/test_step_artifact.json (hunk 2 above)
Comment (mimir-d) [reply json_spec/output/test_step_artifact.json]: done, please close after check

**Review by dhawkes — APPROVED** [review]

**Pull request merged**